include/circt/Dialect/Moore/MooreAttributes.td

```diff
@@ -19,4 +19,10 @@ def FVIntegerAttr : AttrDef<MooreDialect, "FVInteger"> {
   let hasCustomAssemblyFormat = 1;
 }
 
+def FVIntAttr : Attr<CPred<"llvm::isa<FVIntegerAttr>($_self)">,
+                     "arbitrary precision four-valued integer attribute"> {
+  let storageType = [{ circt::moore::FVIntegerAttr }];
+  let returnType = [{ circt::FVInt }];
+}
+
```

Comment:
If you use `FVIntegerAttr` directly in constant op as `(ins FVIntegerAttr:$value)`, then calling `constOp.getValue()` will return the `FVIntegerAttr`. If you use `FVIntAttr` instead, you will get a `getValue()` accessor which returns `FVInt`, and a `getValueAttr()` accessor which returns the `FVIntegerAttr`.

I took the `FVIntAttr` pattern above from the builtin `IntegerAttr`: `hw.constant` has to use `APIntAttr` instead of `Builtin_IntegerAttr` for the same reason. I don't know why this is so obscure in MLIR. Ideally you'd have some way of marking attributes that wrap an underlying type like `APInt` or `FVInt`, and which you'd like to unwrap to that value by default.

Reply:
:partying_face: That works! Really cool, thanks :smiley: